### PR TITLE
Major Requirements Sidebar

### PR DESCRIPTION
**Pull request by NilsBacke**

### CODE REVIEW

File: frontend/src/utils/generate-warnings.ts [hunk 1]
```diff
@@ -184,6 +184,9 @@ function produceUnsatifiedRequirement(
   taken: Map<string, HashableCourse>,
   coursesUsed: Set<string>
 ): IRequirementGroupWarning | undefined {
+  if (!requirementGroup) {
+    return undefined;
+  }
```
Comment (bugsalexander) [inline frontend/src/utils/generate-warnings.ts]: Why did you make this change? I just deployed after removing it, and it worked fine.

**Comment by bugsalexander** [comment]
Also, would it be possible to make the checkmarks green and the X's red?

File: frontend/src/utils/generate-warnings.ts (hunk 1 above)
Comment (NilsBacke) [reply frontend/src/utils/generate-warnings.ts]: You're right! I think I came upon an error before where it was undefined, but I think I was doing something else wrong.

**Comment by bugsalexander** [comment]
The biochem major had a bug – whoops! Yash is fixing it now. 

In other news, courses that are added to the schedule via the + buttons aren't taken into account when producing warnings. Any idea on why?

**Comment by bugsalexander** [comment]
Just merged in master to get the hotfix that @yasz24 pushed for the Biochem major. Now selecting biochem no longer crashes the site!

**Pull request merged**